dialogue-apache-hc5-client/src/main/java/com/palantir/dialogue/hc5/ApacheHttpClientChannels.java

```diff
@@ -235,21 +199,132 @@ static CloseableClient wrap(
                     .clientType(CLIENT_TYPE)
                     .build();
             createMeter.mark();
-            return newInstance;
+            CloseableClient wrapper = new CloseableClientWrapper(newInstance);
+            CleanerSupport.register(wrapper, newInstance::closeApacheClient);
+            return wrapper;
```

Comment:
refactor exists for these three lines -- we must create a delegate which can be strongly referenced by the cleaner without blocking the sentinel reference (the wrapper) from being garbage collected. When the outer class is garbage collected, our cleaner is able to execute.

Reply:
Discussed: I found this kinda hard to reason about, the cleaner stuff always is fun. We discussed instead having private static final InternalCloseableClient and public static final class CloseableClient that takes InternalCloseableClient and delegates. I find that easier to understand, and Carter doesn't so let's just ship this.